packit_service/worker/jobs.py

```diff
@@ -212,97 +91,77 @@ def get_handlers_for_check_rerun(check_name_job: str) -> Set[Type[JobHandler]]:
-    ./tests/unit/test_jobs.py:test_get_config_for_handler_kls
-
-    :param handler_kls: class that will use the JobConfig
-    :param event: which we are reacting to
-    :param package_config: we pick the JobConfig(s) from this package_config instance
-    :return: list of JobConfigs relevant to the given handler and event
-             preserving the order in the config
+    Steve makes sure all the jobs are done with precision.
     """
-    jobs_matching_trigger: List[JobConfig] = get_jobs_matching_event(
-        event=event, package_config=package_config
-    )
-
-    matching_jobs: List[JobConfig] = []
-    for job in jobs_matching_trigger:
-        if handler_kls in MAP_JOB_TYPE_TO_HANDLER[job.type]:
-            matching_jobs.append(job)
 
-    if not matching_jobs:
-        logger.debug("No config found, let's see the jobs that requires this handler.")
-        for job in jobs_matching_trigger:
-            if handler_kls in MAP_REQUIRED_JOB_TYPE_TO_HANDLER[job.type]:
-                matching_jobs.append(job)
+    def __init__(self, event: Optional[Event] = None):
+        self.event = event
+        self._service_config = None
+        log_job_versions()
 
-    if not matching_jobs:
-        logger.warning(
-            f"We did not find any config for {handler_kls} and a following event:\n"
-            f"{event.__class__}"
-        )
+    @property
+    def service_config(self):
+        if self._service_config is None:
+            self._service_config = ServiceConfig.get_service_config()
+        return self._service_config
 
-    return matching_jobs
+    @staticmethod
+    def process_message(event: dict) -> List[TaskResults]:
+        """
+        Entrypoint for message processing.
 
+        Args:
+            event:  dict with webhook/fed-mes payload
 
-def push_initial_metrics(
-    task_accepted_time: datetime,
-    event: Event,
-    handler: JobHandler,
-    number_of_build_targets: Optional[int] = None,
-):
-    pushgateway = Pushgateway()
-    response_time = measure_time(end=task_accepted_time, begin=event.created_at)
-    logger.debug(f"Reporting initial status time: {response_time} seconds.")
-    pushgateway.initial_status_time.observe(response_time)
-    if response_time > 15:
-        pushgateway.no_status_after_15_s.inc()
+        Returns:
+            List of results of the processing tasks.
+        """
+        event_object: Any = Parser.parse_event(event)
 
-    # set the time when the accepted status was set so that we can use it later for measurements
-    event.task_accepted_time = task_accepted_time
+        if not (event_object and event_object.pre_check()):
+            return []
 
-    if isinstance(handler, CoprBuildHandler) and number_of_build_targets:
-        for _ in range(number_of_build_targets):
-            pushgateway.copr_builds_queued.inc()
+        return SteveJobs(event_object).process()
 
-    pushgateway.push()
+    def process(self) -> List[TaskResults]:
```

Comment:
For me it's little bit unclear what `process` means. I get that it's something like `main` method for Steve so maybe... but I won't recommend changing namings since I am not good at this :D

Reply:
I also wasn't able to think about something better :D I am opened to suggestions

Reply:
Isn't this name confusing, b/c it's not clear _what_ is processed (ie. there are no arguments, besides `self`)? If so, that becomes obvious when looking at how it's used, I think: `SteveJobs(event_object).process()`.

How about adding a docstring here, so if someone reads only this method, they can immediately have some context of what is processed, without looking up the usage?